staging/src/k8s.io/client-go/tools/clientcmd/api/v1/types.go

```diff
@@ -89,6 +89,9 @@ type AuthInfo struct {
 	// ClientKeyData contains PEM-encoded data from a client key file for TLS. Overrides ClientKey
 	// +optional
 	ClientKeyData []byte `json:"client-key-data,omitempty"`
+	// ReloadCertFromDisk causes the certificate specified by ClientCertificate and ClientKey to be periodically reloaded from disk.
```

Comment:
Why would we not infer this behavior if ClientKey is set but not ClientKeyData?  That would remove the need for a new flag.

Reply:
That's totally possible. The reason I didn't is so that users need to request this new behaviour. I was concerned we could break some existing setups if we start reloading from disk. But I guess it can go in the release notes.

Reply:
@smarterclayton I have changed such that we default to the rotation behaviour whenever files are provided in rest.Config, meaning that we don't affect kubeconfig parsing at all. We could settle on a middle ground where there's a flag in `rest.Config`, but frankly I think its sensible if long term the data fields are the 'static' way to provide certificates, and providing files means you are accepting that the files will be reloaded.